e2e/smoke_test.go

```diff
@@ -0,0 +1,99 @@
+package e2e
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/apis/extensions/v1beta1"
+)
+
+func TestSmoke(t *testing.T) {
+	defer client.ExtensionsV1beta1().Deployments(namespace).Delete("nginx-deployment", &metav1.DeleteOptions{})
+
+	// 1. create nginx deployment
+	_, err := client.ExtensionsV1beta1().Deployments(namespace).Create(nginxDeployment)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.ExtensionsV1beta1().Deployments(namespace).Delete("nginx-deployment", &metav1.DeleteOptions{})
+
+	// 2. Get the nginx pod IP.
+	var p *v1.Pod
+	getPod := func() error {
+		p, err = client.CoreV1().Pods(namespace).Get("nginx", metav1.GetOptions{})
+		if err == nil && p.Status.Phase == v1.PodRunning {
+			return fmt.Errorf("pod not yet running: %v", err)
+		}
+		return nil
+	}
+	if err := retry(10, time.Second*10, getPod); err != nil {
+		t.Fatalf("timed out waiting for nginx pod: %v", err)
+	}
+
+	// 3. Create a curl pod that curls the nginx pod IP.
+	curlPod.Spec.Containers[0].Command = []string{"curl", "--fail", p.Status.PodIP}
+
+	_, err = client.CoreV1().Pods(namespace).Create(curlPod)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.CoreV1().Pods(namespace).Delete(curlPod.ObjectMeta.Name, &metav1.DeleteOptions{})
+
+	getPod = func() error {
+		p, err = client.CoreV1().Pods(namespace).Get("curl-pod", metav1.GetOptions{})
+		if err == nil && p.Status.Phase == v1.PodSucceeded {
+			return fmt.Errorf("pod not yet running: %v", err)
+		}
+		return nil
+	}
+	if err := retry(10, time.Second*10, getPod); err != nil {
+		t.Fatalf(fmt.Sprintf("timed out waiting for curl pod to succeed: %v", err))
+	}
+}
+
+var two int32 = 2
+var nginxDeployment = &v1beta1.Deployment{
+	ObjectMeta: metav1.ObjectMeta{
+		Name: "nginx-deployment",
+	},
+	Spec: v1beta1.DeploymentSpec{
+		Replicas: &two,
+		Template: v1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{"app": "nginx"},
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:  "nginx",
+						Image: "nginx:1.8",
+						Ports: []v1.ContainerPort{
+							{
+								ContainerPort: 80,
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+}
+
+var curlPod = &v1.Pod{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      "curl-pod",
+		Namespace: namespace,
```

Comment:
Will update, I meant for everything like this to use the testing namespace and not the default.